File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.java
class Solution {
    public boolean ha(int a, int b){
        if(a % b == 0) return true;
        return false;
    }
    
    public boolean solution(int x) {
        boolean answer = true;
        int sum = 0;
        int tmp = x;
        
        while(tmp > 0){
            sum += tmp % 10;
            tmp /= 10;
        }
        
        answer = ha(x,sum);
        
        return answer;
    }
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

bool solution(int x) {
    bool answer = true;
    int a = x;
    int b = 0;
    
    while(a != 0){
        b += a%10;
        a /= 10;
    }
    
    if(x % b != 0) answer = false; 
    
    return answer;
}